app/ui/legacy/src/main/java/com/fsck/k9/ui/account/AccountsViewModel.kt

```diff
@@ -1,8 +1,118 @@
 package com.fsck.k9.ui.account
 
+import android.content.ContentResolver
+import android.database.ContentObserver
+import android.os.Handler
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.fsck.k9.Account
+import com.fsck.k9.AccountsChangeListener
 import com.fsck.k9.Preferences
+import com.fsck.k9.controller.UnreadMessageCountProvider
+import com.fsck.k9.provider.EmailProvider
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.channels.sendBlocking
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.launch
+import timber.log.Timber
 
-class AccountsViewModel(preferences: Preferences) : ViewModel() {
-    val accountsLiveData = AccountsLiveData(preferences)
+@ExperimentalCoroutinesApi
+class AccountsViewModel(
+    val preferences: Preferences,
+    val unreadMessageCountProvider: UnreadMessageCountProvider,
+    private val contentResolver: ContentResolver
+) :
+    ViewModel() {
+
+    private val mutableAccountAndUnreadsLiveData = MutableLiveData<List<DisplayAccount>>()
+
+    val displayAccountsLiveData: LiveData<List<DisplayAccount>>
+        get() = mutableAccountAndUnreadsLiveData
+
+    private val accountsFlow: Flow<List<Account>> =
+        callbackFlow {
+            Timber.d("getAccountsFlow.callbackFlow")
+
+            send(preferences.accounts)
+
+            Timber.d("sent.accounts")
+
+            val accountsChangeListener = AccountsChangeListener {
+                Timber.d("onAccountsChanged")
+                sendBlocking(preferences.accounts)
```

Comment:
`sendBlocking()` can throw an exception. We could either catch and ignore this, or launch a new coroutine in the producer scope and use the suspending send.

```suggestion
                launch {
                    send(preferences.accounts)
                }
```